lib/openxml/shapes/text.rb

```diff
@@ -0,0 +1,73 @@
+require "openxml/elements/shape_non_visual"
+require "openxml/elements/text_body"
+require "openxml/elements/paragraph"
+require "openxml/elements/run"
+require "openxml/elements/text"
+require "openxml/elements/body_properties"
+require "openxml/elements/shape_properties"
+require "openxml/elements/nonvisual_drawing_properties"
+require "openxml/elements/nonvisual_shape_drawing_properties"
+require "openxml/elements/nonvisual_properties"
+
+module OpenXml
+  module Shapes
+    class Text
+      attr_accessor :text
+      private :text=
+
+      def initialize(text)
+        self.text = text
+      end
+
+      def nonvisual_shape_property
+        @nonvisual_shape_property ||= OpenXml::Elements::ShapeNonVisual.new.tap {|nv_shape_property|
+          nv_shape_property << OpenXml::Pptx::Elements::NonvisualDrawingProperties.new.tap { |nvdp|
+            nvdp.id = object_id
+            nvdp.name = "TextBox 1"
+          }
+          nv_shape_property << OpenXml::Pptx::Elements::NonvisualShapeDrawingProperties.new
+          nv_shape_property << OpenXml::Pptx::Elements::NonvisualProperties.new
+        }
+      end
+
+      def shape_property
+        @shape_property = OpenXml::Elements::ShapeProperties.new.tap { |sp|
+          sp << OpenXml::DrawingML::Elements::TransformEffect.new.tap { |effect|
+            effect << OpenXml::DrawingML::Elements::Offset.new.tap { |offset|
+              offset.x = 0
+              offset.y = 0
+            }
+            effect << OpenXml::DrawingML::Elements::Extension.new.tap { |extension|
+              extension.x = 1465545
+              extension.y = 369332
```

Comment:
Fair enough. But I think part of what is giving me pause is that the magic numbers the test is expecting are encoded in essentially two places: the test's expectations, but then also the class itself. For something that's essential to the class and can have sane defaults (like how you construct the non-visual properties, etc. and assign the name "Text Box 1") , that's probably fine. But for the bounds, it makes more sense to have that set by the "consumer" (in this case, the test) just as you had to set the text "Hello World", especially since there's not the same kind of obvious sane default.

In any case, I know you're not leaving this as-is 🙂 . The request for changes was more about getting things correctly namespaced within the gem than this, since I know it's still a WIP.